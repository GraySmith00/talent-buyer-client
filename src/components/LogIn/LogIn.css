.log-in {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 99;
  background-color: rgba(0, 0, 0, 0.5); }
  .log-in .inner-modal {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 20rem;
    width: 60%;
    background-color: rgba(250, 250, 250, 0.8);
    border-radius: 10px; }
