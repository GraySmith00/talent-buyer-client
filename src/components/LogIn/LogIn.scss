.log-in {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 99;
  background-color: rgba(0, 0, 0, 0.5);

  .close-text {
    color: #de0048;
    cursor: pointer;
  }

  .input-wrapper {
    display: flex;
    justify-content: space-between;
    padding: 1em;
    width: 100%;
  }

  .log-in-form {
    margin-top: 3rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
  }

  .login-modal {
    height: 30em;
  }

  .top-inner {
    width: 80%;
    color: #fff;
    font-size: 2.5rem;
  }

  .top-inner h3 {
    margin: 0;
  }

  p {
    margin: 0.8rem 0 0 0;
    color: #fff;
  }

  .top-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    height: 30%;
    width: 100%;
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
      url('./background.jpeg');
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;

    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    -webkit-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
  }
}
