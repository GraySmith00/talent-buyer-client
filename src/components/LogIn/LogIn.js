import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { setUserVenue } from '../../actions/venueActions';
import { logInUser } from '../../actions/userActions';
import { populateUserOffers } from '../../actions/offerActions';
import { populateUserWatchlist } from '../../actions/watchlistActions';

import ModalButton from '../styledComponents/ModalButton';
import InputField from '../styledComponents/InputField';
import ModalForm from '../styledComponents/ModalForm';
import InnerModal from '../styledComponents/InnerModal';
import './LogIn.css';

export class LogIn extends Component {
  state = {
    email: '',
    password: ''
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  toggleModal = () => {
    this.props.closeLogInModal();
    this.props.openSignUpModal();
  };

  handleSubmit = async e => {
    const { email, password } = this.state;
    e.preventDefault();
    const userCreds = {
      buyer: {
        email,
        password
      }
    };
    const loggedInUser = await this.props.logInUser(userCreds);
    const userVenue = await this.props.setUserVenue(loggedInUser.id);
    await this.props.populateUserOffers(loggedInUser.id);
    await this.props.populateUserWatchlist();

    if (loggedInUser && userVenue) {
      this.props.history.push('/home');
    }
  };

  render() {
    return (
      <div className="log-in">
        <InnerModal className="login-modal">
          <div className="top-container">
            <div className="top-inner">
              <h3 className="header-text">Welcome to Talent Buyer!</h3>
            </div>
          </div>
          <ModalForm className="log-in-form" onSubmit={this.handleSubmit}>
            <p>Log In With Email</p>
            <InputField
              type="email"
              name="email"
              value={this.state.email}
              placeholder="email"
              onChange={this.handleChange}
            />
            <InputField
              type="password"
              name="password"
              value={this.state.password}
              placeholder="password"
              onChange={this.handleChange}
            />
            <ModalButton>Submit</ModalButton>
            <p onClick={this.toggleModal} className="close-text">
              Need to Sign Up?
            </p>
          </ModalForm>
        </InnerModal>
      </div>
    );
  }
}

LogIn.propTypes = {
  closeLogInModal: PropTypes.func.isRequired,
  openSignUpModal: PropTypes.func,
  logInUser: PropTypes.func.isRequired,
  setUserVenue: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  populateUserOffers: PropTypes.func.isRequired,
  populateUserWatchlist: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => ({
  logInUser: userCreds => dispatch(logInUser(userCreds)),
  setUserVenue: userId => dispatch(setUserVenue(userId)),
  populateUserOffers: userId => dispatch(populateUserOffers(userId)),
  populateUserWatchlist: () => dispatch(populateUserWatchlist())
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(LogIn)
);
