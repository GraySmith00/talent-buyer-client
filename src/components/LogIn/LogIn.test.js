import React from 'react';
import { shallow } from 'enzyme';

import LogIn from './LogIn';

describe('LogIn component', () => {
  let wrapper;
  let mockCloseLoginModal;

  beforeEach(() => {
    mockCloseLoginModal = jest.fn();
    wrapper = shallow(<LogIn closeLogInModal={mockCloseLoginModal} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
