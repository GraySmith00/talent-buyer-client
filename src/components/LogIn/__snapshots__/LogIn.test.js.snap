// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn
    closeLogInModal={[Function]}
    history={
        Object {
            "action": "PUSH",
            "length": 50,
            "location": Object {
              "hash": "",
              "key": "hvocjm",
              "pathname": "/home",
              "search": "",
            },
          }
    }
    logInUser={[Function]}
    populateUserOffers={[Function]}
    populateUserWatchlist={[Function]}
    setUserVenue={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        className="login-modal"
>
        <div
                className="top-container"
        >
                <div
                        className="top-inner"
                >
                        <h3
                                className="header-text"
                        >
                                Welcome to Talent Buyer!
                        </h3>
                </div>
        </div>
        <Unknown
                className="log-in-form"
                onSubmit={[Function]}
        >
                <p>
                        Log In With Email
                </p>
                <Unknown
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
                />
                <Unknown
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
                />
                <Unknown>
                        Submit
                </Unknown>
                <p
                        className="close-text"
                        onClick={[Function]}
                >
                        Need to Sign Up?
                </p>
        </Unknown>
</Unknown>,
      "className": "log-in",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="top-container"
>
            <div
                        className="top-inner"
            >
                        <h3
                                    className="header-text"
                        >
                                    Welcome to Talent Buyer!
                        </h3>
            </div>
</div>,
          <Unknown
            className="log-in-form"
            onSubmit={[Function]}
>
            <p>
                        Log In With Email
            </p>
            <Unknown
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
            <Unknown
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <Unknown>
                        Submit
            </Unknown>
            <p
                        className="close-text"
                        onClick={[Function]}
            >
                        Need to Sign Up?
            </p>
</Unknown>,
        ],
        "className": "login-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="top-inner"
>
              <h3
                            className="header-text"
              >
                            Welcome to Talent Buyer!
              </h3>
</div>,
            "className": "top-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3
                className="header-text"
>
                Welcome to Talent Buyer!
</h3>,
              "className": "top-inner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Talent Buyer!",
                "className": "header-text",
              },
              "ref": null,
              "rendered": "Welcome to Talent Buyer!",
              "type": "h3",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Log In With Email
</p>,
              <Unknown
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
              <Unknown
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <Unknown>
                Submit
</Unknown>,
              <p
                className="close-text"
                onClick={[Function]}
>
                Need to Sign Up?
</p>,
            ],
            "className": "log-in-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In With Email",
              },
              "ref": null,
              "rendered": "Log In With Email",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 80%;
  outline: none;
  border: none;
  margin-top: 10px;
  padding: 5px 10px;
  font-size: 1.2rem;
  background-color: rgba(250, 250, 250, 0.3);
  color: #fff;
  border-radius: 5px;
",
                  ],
                },
                "displayName": "styled.input",
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 80%;
  outline: none;
  border: none;
  margin-top: 10px;
  padding: 5px 10px;
  font-size: 1.2rem;
  background-color: rgba(250, 250, 250, 0.3);
  color: #fff;
  border-radius: 5px;
",
                  ],
                },
                "displayName": "styled.input",
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "rules": Array [
                    "
  margin-top: 15px;
  width: 80%;
  padding: 15px;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #D52F7F;
  color: #fff;
  cursor: pointer;
",
                  ],
                },
                "displayName": "styled.button",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Need to Sign Up?",
                "className": "close-text",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Need to Sign Up?",
              "type": "p",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
 display: flex;
 flex-direction: column;
 align-items: center;
 width: 100%;
",
              ],
            },
            "displayName": "styled.form",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "form",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
display: flex;
flex-direction: column;
align-items: center;
height: 40rem;
width: 30rem;
background-color: #21222b;
border-radius: 10px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          className="login-modal"
>
          <div
                    className="top-container"
          >
                    <div
                              className="top-inner"
                    >
                              <h3
                                        className="header-text"
                              >
                                        Welcome to Talent Buyer!
                              </h3>
                    </div>
          </div>
          <Unknown
                    className="log-in-form"
                    onSubmit={[Function]}
          >
                    <p>
                              Log In With Email
                    </p>
                    <Unknown
                              name="email"
                              onChange={[Function]}
                              placeholder="email"
                              type="email"
                              value=""
                    />
                    <Unknown
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="password"
                              value=""
                    />
                    <Unknown>
                              Submit
                    </Unknown>
                    <p
                              className="close-text"
                              onClick={[Function]}
                    >
                              Need to Sign Up?
                    </p>
          </Unknown>
</Unknown>,
        "className": "log-in",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="top-container"
>
              <div
                            className="top-inner"
              >
                            <h3
                                          className="header-text"
                            >
                                          Welcome to Talent Buyer!
                            </h3>
              </div>
</div>,
            <Unknown
              className="log-in-form"
              onSubmit={[Function]}
>
              <p>
                            Log In With Email
              </p>
              <Unknown
                            name="email"
                            onChange={[Function]}
                            placeholder="email"
                            type="email"
                            value=""
              />
              <Unknown
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="password"
                            value=""
              />
              <Unknown>
                            Submit
              </Unknown>
              <p
                            className="close-text"
                            onClick={[Function]}
              >
                            Need to Sign Up?
              </p>
</Unknown>,
          ],
          "className": "login-modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="top-inner"
>
                <h3
                                className="header-text"
                >
                                Welcome to Talent Buyer!
                </h3>
</div>,
              "className": "top-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="header-text"
>
                  Welcome to Talent Buyer!
</h3>,
                "className": "top-inner",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to Talent Buyer!",
                  "className": "header-text",
                },
                "ref": null,
                "rendered": "Welcome to Talent Buyer!",
                "type": "h3",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Log In With Email
</p>,
                <Unknown
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="email"
                  value=""
/>,
                <Unknown
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
/>,
                <Unknown>
                  Submit
</Unknown>,
                <p
                  className="close-text"
                  onClick={[Function]}
>
                  Need to Sign Up?
</p>,
              ],
              "className": "log-in-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log In With Email",
                },
                "ref": null,
                "rendered": "Log In With Email",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "email",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "rules": Array [
                      "
  width: 80%;
  outline: none;
  border: none;
  margin-top: 10px;
  padding: 5px 10px;
  font-size: 1.2rem;
  background-color: rgba(250, 250, 250, 0.3);
  color: #fff;
  border-radius: 5px;
",
                    ],
                  },
                  "displayName": "styled.input",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "rules": Array [
                      "
  width: 80%;
  outline: none;
  border: none;
  margin-top: 10px;
  padding: 5px 10px;
  font-size: 1.2rem;
  background-color: rgba(250, 250, 250, 0.3);
  color: #fff;
  border-radius: 5px;
",
                    ],
                  },
                  "displayName": "styled.input",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "rules": Array [
                      "
  margin-top: 15px;
  width: 80%;
  padding: 15px;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #D52F7F;
  color: #fff;
  cursor: pointer;
",
                    ],
                  },
                  "displayName": "styled.button",
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Need to Sign Up?",
                  "className": "close-text",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Need to Sign Up?",
                "type": "p",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
 display: flex;
 flex-direction: column;
 align-items: center;
 width: 100%;
",
                ],
              },
              "displayName": "styled.form",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "form",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
display: flex;
flex-direction: column;
align-items: center;
height: 40rem;
width: 30rem;
background-color: #21222b;
border-radius: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
